salt/modules/aixpkg.py

```diff
@@ -392,10 +664,77 @@ def upgrade_available(name, **kwargs):
     """
     Check whether or not an upgrade is available for a given package
 
+    .. versionchanged:: 3005
+
     CLI Example:
 
     .. code-block:: bash
 
         salt '*' pkg.upgrade_available <package name>
+
+    Note: currently only functional for rpm packages due to filesets do not have a specific location to check
+        Requires yum of dnf available in order to query a repository
+
     """
-    return latest_version(name) != ""
+    # AIX packaging includes info on filesets and rpms
+    rpm_found = False
+    version_found = ""
+
+    libpathenv = {"LIBPATH": "/opt/freeware/lib:/usr/lib"}
+    if pathlib.Path("/opt/freeware/bin/dnf").is_file():
```

Comment:
**suggestion:** It looks like you are re-using code to detect the command path. Is it possible to move this logic into a function?

Reply:
I thought about doing that, but the parameters for actions are different when the check for which type of of yum and dnf is checked, e.g. yum being Py 2 based verses yum being Py 3 based and possibly a symbolic link into dnf and additional options.

I agree with you, not happy about the similar code getting repeated